packages/plugin-hardhat/src/deploy-proxy.ts

```diff
@@ -73,6 +84,13 @@ export function makeDeployProxy(hre: HardhatRuntimeEnvironment): DeployFunction
     const inst = ImplFactory.attach(proxyDeployment.address);
     // @ts-ignore Won't be readonly because inst was created through attach.
     inst.deployTransaction = proxyDeployment.deployTransaction;
+    if (opts.platform && proxyDeployment.deploymentId !== undefined) {
+      inst.deployed = async () => {
+        assert(proxyDeployment.deploymentId !== undefined);
+        await waitForDeployment(hre, opts, inst.address, proxyDeployment.deploymentId);
+        return inst;
```

Comment:
Is there a chance we might want to update `inst.deployTransaction` here?

Should we also add a call to the original `inst.deployed()` at the end?